src/io/flutter/vmService/DisplayRefreshRateManager.java

```diff
@@ -78,20 +78,22 @@ public Double getCurrentDisplayRefreshRateRaw() {
   }
 
   private CompletableFuture<Double> getDisplayRefreshRate() {
-    final double unknownRefreshRate = 0.0;
-
-    final String flutterViewId = vmServiceManager.getFlutterViewId().exceptionally(exception -> {
+    final CompletableFuture<Double> displayRefreshRate = new CompletableFuture<Double>();
+    vmServiceManager.getFlutterViewId().exceptionally(exception -> {
       // We often see "java.lang.RuntimeException: Method not found" from here; perhaps a race condition?
       LOG.warn(exception.getMessage());
       return null;
-    }).join();
-
-    if (flutterViewId == null) {
-      // Fail gracefully by returning the default.
-      return CompletableFuture.completedFuture(defaultRefreshRate);
-    }
-
-    return invokeGetDisplayRefreshRate(flutterViewId);
+    }).whenComplete((String id, Throwable throwable) -> {
+      if (id == null) {
+        // Fail gracefully by returning the default.
+        displayRefreshRate.complete(defaultRefreshRate);
+      } else {
+        invokeGetDisplayRefreshRate(id).whenComplete((Double refreshRate, Throwable t) -> {
+          displayRefreshRate.complete(refreshRate);
```

Comment:
If `t` is not null here, should we instead complete `displayRefreshRate` with the `defaultRefreshRate`?

Reply:
Added exception handling via `exceptionally()`

Reply:
I think you either want to use `thenApplyAsync()` and `exceptionally()`, or use `whenComplete()`. It looks like whenComplete gives you both the value, if it completes nornally or the exception if it fails.

Reply:
done.